src/app.js

```diff
@@ -84,27 +70,35 @@ export default function (app) {
             hooks.onAction[i](name, data)
           }
 
-          if (shouldRender) {
-            var oldModel = model
-            model = merge(model, action(model, data))
+          var result = action(model, data, actions, onError)
 
+          if (result === undefined || typeof result.then === "function") {
+            return result
+
+          } else {
             for (i = 0; i < hooks.onUpdate.length; i++) {
-              hooks.onUpdate[i](oldModel, model, data)
+              hooks.onUpdate[i](model, result, data)
             }
-
+            model = merge(model, result)
             render(model, view)
-
-            return actions
-          } else {
-            return action(model, actions, data, onError)
           }
         }
       } else {
-        init(container[key], action, shouldRender, name)
+        init(container[key], action, name)
       }
     })
   }
 
+  load(function () {
+    root = app.root || document.body.appendChild(document.createElement("div"))
+
+    render(model, view)
+
+    for (var i = 0; i < subscriptions.length; i++) {
+      subscriptions[i](model, actions, onError)
+    }
+  })
+
   function load(fn) {
```

Comment:
@liadbiz Thanks! :) 

See #69 thread for the original reason we introduced this code.